packages/wrangler/src/api/startDevWorker/DevEnv.ts

```diff
@@ -0,0 +1,145 @@
+import { EventEmitter } from "node:events";
+// import { readFileSync } from "../../parse";
+import { BundlerController } from "./BundlerController";
+import { ConfigController } from "./ConfigController";
+import { LocalRuntimeController } from "./LocalRuntimeController";
+import { ProxyController } from "./ProxyController";
+import { RemoteRuntimeController } from "./RemoteRuntimeController";
+import type { RuntimeController } from "./BaseController";
+import type { ErrorEvent } from "./events";
+import type { StartDevWorkerOptions, DevWorker } from "./types";
+
+/**
+ * @internal
+ */
+export class DevEnv extends EventEmitter {
```

Comment:
This class contains the glue code for all startDevWorker controllers. This is the contract between all controllers. You _could_ override the instances of each/any controller in the constructor but we're not exposing this to users yet until we're more sure of the contract.